objects-revisited/main.ts

```diff
@@ -0,0 +1,89 @@
+interface Employee {
+  position: string;
+  daysOfWeekWorking: string[];
+  fullTime?: boolean;
+}
+
+interface Business {
+  opens: string;
+  closes: string;
+  totalEmployees: number;
+  daysOpen: string[];
+  employees: Record<string, Employee>;
+}
+
+function checkIfFullTime(daysWorking: string[]): boolean {
+  return daysWorking.length >= 5;
+}
+
+async function addEmployees(): Promise<void> {
+  try {
+    const response = await fetch('https://jsonplaceholder.typicode.com/users');
+    if (!response.ok) throw new Error(`Error: ${response.status}`);
+    const users = await response.json();
+    for (let i = 0; i < 4; i++) {
+      const randomJob = jobs[Math.round(Math.random() * (jobs.length - 1))];
+      const randomDays: string[] = [];
+      const randomNumOfDays = Math.floor(Math.random() * 5) + 1;
+      const weekdays = ['Mon', 'Tue', 'Wed', 'Thu', 'Fri'];
+      for (let i = 0; i < randomNumOfDays; i++) {
+        const randomIndex = Math.floor(Math.random() * weekdays.length);
+        randomDays.push(weekdays.splice(randomIndex, 1)[0]);
+      }
+      randomDays.push('Sat', 'Sun');
+      const fullTimeStatus = checkIfFullTime(randomDays);
+      business.employees[users[i].name] = {
+        position: randomJob,
+        daysOfWeekWorking: randomDays,
+        fullTime: fullTimeStatus,
+      };
+      business.totalEmployees++;
+    }
```

Comment:
After you add the base employees you can move checking if an employee is full time to your DOMContentLoaded so that you are only checking after you have all of the employees who is full time or not and can avoid having multiple checks for it.